tests/lib/processor.js

```diff
@@ -726,6 +726,26 @@ describe("processor", () => {
 
         });
 
+        it("should preserve messages without valid `line` property", () => {
+
+            // change cache for `postprocess`
+            processor.preprocess([
+                "```ts",
+                "const num: number = 0;",
+                "```"]
+                .join("\n"));
+
+            const message = { message: "Parsing error: \"parserOptions.project\" has been set for @typescript-eslint/parser.", ruleId: null };
+            const result = processor.postprocess([[message]]);
+
+            assert.strictEqual(result.length, 1);
+            assert.deepStrictEqual(result[0], {
+                ...message,
+                line: 1,
+                column: 1
+            });
+        });
```

Comment:
Some future change might hard-code line 1 column 1, breaking the intended behavior but still passing this test. Let's improve the test to better protect the intended behavior. There are already three code blocks in the test code, so maybe we can reuse that, and it turns out it's easy to pass this message to multiple code blocks! How's this?

```suggestion
        it("should attach messages without `line` to opening code fence", () => {
            const message = { message: "Parsing error: \"parserOptions.project\" has been set for @typescript-eslint/parser.", ruleId: null };
            const result = processor.postprocess([[message], [message], [message]]);

            assert.strictEqual(result.length, 3);
            assert.deepStrictEqual(result[0], {
                ...message,
                line: 3,
                column: 1
            });
            assert.deepStrictEqual(result[1], {
                ...message,
                line: 14,
                column: 4
            });
            assert.deepStrictEqual(result[2], {
                ...message,
                line: 23,
                column: 3
            });
        });
```